src/templates/plugin.routing.js

```diff
@@ -55,16 +55,29 @@ function localeRoute (route, locale) {
     const isDefaultLocale = locale === defaultLocale
     // if route has a path defined but no name, resolve full route using the path
     const isPrefixed =
-        // don't prefix default locale
-        !(isDefaultLocale && [STRATEGIES.PREFIX_EXCEPT_DEFAULT, STRATEGIES.PREFIX_AND_DEFAULT].includes(strategy)) &&
-        // no prefix for any language
-        !(strategy === STRATEGIES.NO_PREFIX) &&
-        // no prefix for different domains
-        !i18n.differentDomains
-
-    let path = (isPrefixed ? `/${locale}${route.path}` : route.path)
-    path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
-    localizedRoute.path = path
+      // don't prefix default locale
+      !(isDefaultLocale && [STRATEGIES.PREFIX_EXCEPT_DEFAULT, STRATEGIES.PREFIX_AND_DEFAULT].includes(strategy)) &&
+      // no prefix for any language
+      !(strategy === STRATEGIES.NO_PREFIX) &&
+      // no prefix for different domains
+      !i18n.differentDomains
+    const thisRoute = this.router.resolve(route.path).route
+    const routeName = this.getRouteBaseName(thisRoute)
+
+    if (isPrefixed && !routeName) {
+      let path = `/${locale}${route.path}`
+      path = path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
+      localizedRoute.path = path
+    } else {
+      localizedRoute.path = route.path.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'
```

Comment:
Those two `replace` calls are basically the same so you can re-arrange the code a bit to not repeat yourself.

Reply:
no, the first path includes locale, other does not

Reply:
I'm talking about the `.replace(/\/+$/, '') + (trailingSlash ? '/' : '') || '/'` part.

You can refactor the code so that you don't repeat that part.

Reply:
it's done.